alerts/config.go

```diff
@@ -29,29 +41,33 @@ type Config struct {
 func (c Config) Validate(validator structure.Validator) {
 	validator.String("UserID", &c.UserID).Using(user.IDValidator)
 	validator.String("FollowedID", &c.FollowedID).Using(user.IDValidator)
-	if c.Low != nil {
-		c.Low.Validate(validator)
+	c.Alerts.Validate(validator)
+}
+
+func (i Alerts) Validate(validator structure.Validator) {
+	if i.Low != nil {
+		i.Low.Validate(validator)
 	}
-	if c.UrgentLow != nil {
-		c.UrgentLow.Validate(validator)
+	if i.UrgentLow != nil {
+		i.UrgentLow.Validate(validator)
 	}
-	if c.High != nil {
-		c.High.Validate(validator)
+	if i.High != nil {
+		i.High.Validate(validator)
 	}
-	if c.NotLooping != nil {
-		c.NotLooping.Validate(validator)
+	if i.NotLooping != nil {
+		i.NotLooping.Validate(validator)
 	}
-	if c.NoCommunication != nil {
-		c.NoCommunication.Validate(validator)
+	if i.NoCommunication != nil {
+		i.NoCommunication.Validate(validator)
 	}
 }
 
 // Base describes the minimum specifics of a desired alert.
 type Base struct {
 	// Enabled controls whether notifications should be sent for this alert.
-	Enabled bool
+	Enabled bool `json:"enabled"`
 	// Repeat is measured in minutes.
-	Repeat DurationMinutes `json:"repeat"`
+	Repeat DurationMinutes `json:"repeat,omitempty"`
```

Comment:
I believe I understand what you're saying. However, in this case, the omitempty only affects the JSON representation. The BSON representation seen by Mongo does not have the omitempty flag, and so I believe should be OK here. I'll write a test to confirm.

That said, I hadn't considered the effect of omitempty on a Mongo upsert operation, so I should review the bson struct tags for that behavior in other places. A few tests would also be good, to prevent regressions or unexpected behavior.